### ISSUE TITLE
API review: Remove ServiceDescriber

### ISSUE THREAD

**Issue by muratg** [aspnet/DependencyInjection#193]
Describer work needs to go to hosting. Need better hosting configuration story.  Assigning to @HaoK per API review meeting.

**Comment by muratg** [comment]
@HaoK Do we need to file Hosting items as well?

**Comment by HaoK** [comment]
Needed until MVC functionals are updated to no longer use this

**Comment by GrabYourPitchforks** [comment]
There are tons of useful factory methods hanging off `ServiceDescriber`. If we get rid of the type, I propose moving these to static methods on `ServiceDescriptor` instead.

**Comment by HaoK** [comment]
I already copied all of them to descriptor as statics. so that is already done as part of the initial push.

**Comment by davidfowl** [comment]
@HaoK that code should all be in interfaces. The main package should serve purely as our default implementation.

**Comment by GrabYourPitchforks** [comment]
Not sure how I missed that earlier. Thanks @haok!

**Comment by HaoK** [comment]
@davidfowl yeah that's where they are there: its very hard to differentiate between ServiceDescriPTOR vs ServiceDescriBER, but right now they both have the same sugar https://github.com/aspnet/DependencyInjection/blob/dev/src/Microsoft.Framework.DependencyInjection.Interfaces/ServiceDescriptor.cs#L81

**Comment by davidfowl** [comment]
I am blind.

**Comment by davidfowl** [comment]
@HaoK Who is assigned to update the MVC functional tests?

**Comment by HaoK** [comment]
I still need to file an issue for that, they are blocked until I get the HostingServices.Create changes in, so I'll file an issue in MVC once I push that PR and assign it to @rynowak to switch MVC functionals over

**Comment by davidfowl** [comment]
We should try to get those changes in asap.

**Comment by davidfowl** [comment]
Where are we on this one? /cc @dougbu

**Comment by dougbu** [comment]
working on it.  down to a couple of hundred failing functional tests (from over a thousand).  debugging the order things run and their precedence...

@davidfowl / @HaoK stop by when you get in.  I may be doing exactly the Wrong Thing:tm:

**Comment by dougbu** [comment]
see PRs aspnet/Hosting#184 (now commits aspnet/Hosting@89a1ec1 through aspnet/Hosting@ee55e38) and aspnet/Mvc#2137.  those should enable completion of this work in aspnet/DependencyInjection.

**Comment by muratg** [comment]
Thanks @dougbu. @HaoK are there any other blockers?

**Comment by HaoK** [comment]
I don't believe so should be good to go now.

**Comment by muratg** [comment]
:+1:

**Comment by dougbu** [comment]
aspnet/Mvc@cc7b319 now in.  should unblock fixing this issue.

**Comment by davidfowl** [comment]
Ok time to nuke it

**Comment by muratg** [comment]
Cool, @HaoK can you get it in?

**Comment by HaoK** [comment]
Yah I'll carpet bomb in a bunch of easy cleanup changes on my plate today now that builds are stable.
